### ISSUE TITLE
Update function configuration request doesn't change executionRole and memorySize value

### ISSUE THREAD

**Issue by sebarys** [localstack/localstack#3282]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

When using java-aws-sdk `UpdateFunctionConfigurationRequest` should
- change function execution role when `role` field is set
- change function memory size when `memorySize` field is set

When I tried such operations against localstack container in version `localstack/localstack:0.11.2` those operations didn't publish new function version.

## Actual behavior

When `UpdateFunctionConfigurationRequest` are having different than actual values for `role` or `memorySize`, those fields in function configuration should be changed.

# Steps to reproduce
- create function
- perform `UpdateFunctionConfigurationRequest` with different than actual values for `role` or `memorySize`
- verify if values changed in function configuration

## Command used to start LocalStack

Used testcontainers java localstack library (`localstack/localstack:0.11.2`).

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
```
UpdateFunctionConfigurationRequest.builder
      .functionName(functionName)
      .runtime(runtimeName)
      .handler(handlerName)
      .layers(functionLayers.asJava)
      .role(lambdaExecutionRole)
      .memorySize(memorySize)
      .build()

lambdaClient.updateFunctionConfiguration(updateFunctionConfigurationRequest)
```

**Comment by whummer** [comment]
Thanks for reporting @sebarys . This should be fixed in #3295 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks

**Comment by sebarys** [comment]
Thanks @whummer @macnev2013 - I've tested version `localstack/localstack:0.12.3` and it works now as expected!
